package/src/components/ChannelPreview/ChannelPreview.tsx

```diff
@@ -12,120 +13,40 @@ import { ChatContextValue, useChatContext } from '../../contexts/chatContext/Cha
 
 import type { DefaultStreamChatGenerics } from '../../types/types';
 
-export type ChannelPreviewPropsWithContext<
+export type ChannelPreviewProps<
   StreamChatGenerics extends DefaultStreamChatGenerics = DefaultStreamChatGenerics,
-> = Pick<ChatContextValue<StreamChatGenerics>, 'client'> &
-  Pick<ChannelsContextValue<StreamChatGenerics>, 'Preview' | 'forceUpdate'> & {
+> = Partial<Pick<ChatContextValue<StreamChatGenerics>, 'client'>> &
+  Partial<Pick<ChannelsContextValue<StreamChatGenerics>, 'Preview' | 'forceUpdate'>> & {
     /**
      * Instance of Channel from stream-chat package.
      */
     channel: Channel<StreamChatGenerics>;
   };
 
-/**
- * This component manages state for the ChannelPreviewMessenger UI component and receives
- * all props from the ChannelListMessenger component.
- */
-const ChannelPreviewWithContext = <
+export const ChannelPreview = <
   StreamChatGenerics extends DefaultStreamChatGenerics = DefaultStreamChatGenerics,
 >(
-  props: ChannelPreviewPropsWithContext<StreamChatGenerics>,
+  props: ChannelPreviewProps<StreamChatGenerics>,
 ) => {
-  const { channel, client, forceUpdate: channelListForceUpdate, Preview } = props;
+  const { channel, client: propClient, forceUpdate: propForceUpdate, Preview: propPreview } = props;
```

Comment:
Why do we allow the `client`, which is meant to be a singleton get overriden for `ChannelPreview` specifically ? Seems sketchy, but if there's a good reason for it it's fine

Reply:
It's just because of the relic that the component has the prop. If we don't use it from prop there is no good reason of having the prop but then removing it will be a breaking change.

Reply:
Ah, I see. I mean I don't really like it but I suppose we can stick with it for now and remove this later on (V7 maybe).

Reply:
Yep

Reply:
But no, wait - it does come as a prop but it gets passed by the overarching `ChannelPreview` component. Which means, whoever uses it does not really pass the `client` at all, but rather it's inferred from the context anyway. So, in this case `propClient` and `contextClient` would always be the same right ?

Reply:
The context would be prioritized, yes, since propClient will be undefined. But that doesn't stop anyone to not pass client through prop if they would like to.

Reply:
Same goes for all other props that are being handled here both from the context and within this component.

Reply:
Yes

Reply:
The removal of props is a breaking change that is why i didn't remove them in the PR. I hope you get that as well.

Reply:
Okay, this should be fine then.